### PR TITLE
[Azure] [OpenAI] Recalculate event.duration

### LINKED ISSUE #9987: [Azure] [OpenAI] event.duration is fetched as negative from Azure
Event duration should be the latency of the event, however in some cases it comes in as a negative value (specifically from events which generate images from Dall-E 3):

We should also double check `event.duration` to be accurate as approximately `azure.open_ai.properties.response_time - azure.open_ai.properties.request_time`.

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message

Recalculates `event.duration` by using response.time - request.time.

`event.duration` comes in as negative and wrongly calculated (for logs, metrics are fine) directly from Azure. 

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-  Closes #9987

### CODE REVIEW

File: packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -78,6 +78,17 @@ processors:
       field: azure.open_ai.duration_ms
       target_field: event.duration
       ignore_missing: true
+  - script:
+      lang: painless
+      source: |
+        if (ctx.azure?.open_ai.properties instanceof Map) {
```
Comment (muthu-mps) [inline packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml:84]:
```suggestion
        if (ctx.azure?.open_ai?.properties instanceof Map) {
```

File: packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -78,6 +78,17 @@ processors:
...
+          if (ctx.azure.open_ai.properties.model_name == "dall-e-3" && ctx.event.duration != null) {
```
Comment (muthu-mps) [inline packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml:85]: Are we expecting the model name for image generation as `dall-e-3` and not other possible model names for image generation?

File: packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -78,6 +78,17 @@ processors:
       field: azure.open_ai.duration_ms
       target_field: event.duration
       ignore_missing: true
+  - script:
```
Comment (muthu-mps) [inline packages/azure_openai/data_stream/logs/elasticsearch/ingest_pipeline/default.yml:81]: Please add tag to this script processor. Also add a comment on why the calculation has been done.

**Comment by shmsr** [comment]
Shouldn't we consider dall-e-2 as well?

See: https://platform.openai.com/docs/api-reference/images/create

**Comment by muthu-mps** [comment]
> Shouldn't we consider dall-e-2 as well?
> 
> See: https://platform.openai.com/docs/api-reference/images/create

Yes, @shmsr - In my previous comment, I have asked to check the model names we need to include for the image generation which falls under this category.

**Comment by muthu-mps** [comment]
@lucian-ioan - We cannot filter by image generation API. It happens with chat completions model as well. We need to get more clarity on what scenario this happens. 

event duration

**Comment by lucian-ioan** [comment]
@muthu-mps @shmsr 

Based on your comments, we cannot filter specifically for dall-e-3. So instead (as Muthu suggested), we can recalculate event.duration for every log while keeping the original just in case.

Let me know what you think.

**Comment by muthu-mps** [comment]
> @muthu-mps @shmsr
> 
> Based on your comments, we cannot filter specifically for dall-e-3. So instead, we can recalculate event.duration for every log while keeping the original just in case (as Muthu suggested).
> 
> Let me know what you think.

As the response is not consistent across models from the Azure service itself. Lets wait for updates and investigate on how the negative value is received in the duration. Lets continue to use the formula in the Kibana dashboards for now.

**Pull request closed without merging**